lib/k8s-e2e-tests/tests/vector-agent.rs

```diff
@@ -537,10 +604,29 @@ async fn pod_metadata_annotation() -> Result<(), Box<dyn std::error::Error>> {
         )
         .await?;
 
-    let mut log_reader = framework.logs(&namespace, &format!("daemonset/{}", override_name))?;
+    // Make sure we read the correct nodes logs.
+    let node = framework
+        .get_node_for_pod(&pod_namespace, "test-pod")
+        .await
+        .expect("need the node name");
+
+    let vector_pod = framework
+        .get_pod_on_node(&namespace, &node, &override_name)
+        .await
+        .expect("cant get the vector pod running on the test node");
+
+    let mut log_reader = framework.logs(&namespace, &format!("pod/{}", vector_pod))?;
     smoke_check_first_line(&mut log_reader).await;
     let k8s_version = framework.kubernetes_version().await?;
-    let minor = u8::from_str(&k8s_version.minor()).expect("Couldn't get u8 from String!");
+
+    // Replace all non numeric chars from the version number
+    let numeric_regex = regex::Regex::new(r#"[^\d]"#).unwrap();
+    let minor = k8s_version.minor();
+    let numeric_minor = numeric_regex.replace(&minor, "");
+    let minor = u8::from_str(&numeric_minor).expect(&format!(
+        "Couldn't get u8 from String, received {} instead!",
+        k8s_version.minor()
+    ));
```

Comment:
What prompted this change? Was it an error we hadn't caught before?

Reply:
I was getting a minor version of `19+`, which was not completely numeric.

Reply:
Funny way to denote a minor version. That's like saying "you might have 19, or 20, or 21, or ..." 😄

Reply:
Perhaps they mean it's just a little bit better than 19.

```
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.8-eks-96780e", GitCommit:"96780e1b30acbf0a52c38b6030d7853e575bcdf3", GitTreeState:"clean", BuildDate:"2021-03-10T21:32:29Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```